src/resolvers/products/index.ts

```diff
@@ -1,10 +1,78 @@
 import { generateId, EntityType } from '../../functions/generate-binary-id';
-import { CreateProductInput } from '../../types/products-types';
+import {
+  CreateProductInput,
+  ProductQueryParams,
+  LooseObject,
+  Product,
+} from '../../types/products-types';
 import ProductModel from '../../models/products';
 import { Context } from 'koa';
 import { Account } from '../../types/accounts-types';
+import r from 'ramda';
+import { covertToQueryFilter } from '../helpers/convert-to-query';
 
 export const ProductResolver = {
+  Query: {
+    products: async (_: never, data: ProductQueryParams, ctx: Context) => {
+      let {
+        first = 5,
+        after,
+        filter,
+        sort = {
+          name: 1,
+        },
+      } = data;
+      let product: Array<Product> = [];
+      let finalData: LooseObject = {};
+
+      const options: LooseObject = {
+        limit: parseInt(first.toString()),
+        sort,
+      };
+
+      options.sort.name = options.sort.name == 1 ? `asc` : `desc`;
+      const queryFilter = covertToQueryFilter(filter);
+
+      if (!after) {
+        options.sort['createdAt'] = `asc`;
+        product = await ProductModel.find({}, null, options).where(
+          queryFilter!,
+        );
+      }
+
+      if (after) {
+        product = await ProductModel.find(
+          {
+            cursor: { $gte: after },
+          },
+          null,
+          options,
+        ).where(queryFilter!);
+      }
+
+      if (product.length == 0) throw new Error(`No products found.`);
```

Comment:
It should not throw an error if no products are found but rather return edges with an empty array.